packages/neuron-wallet/tests/models/transaction-size.test.ts

```diff
@@ -25,12 +25,26 @@ describe('TransactionSize', () => {
 
   it('witnessArgs', () => {
     const result = TransactionSize.witness(witnessArgs)
-    expect(result).toEqual(HexUtils.byteLength('0x10000000100000001000000010000000') + 4 + 4)
+    // FIXME: this may cause bug if not used properly
+    //
+    // in `@ckb-lumos/codec` format:
+    // 0x1400000010000000100000001400000000000000 for
+    // {
+    //  lock:undefined
+    //  input_type:"0x" // more 4 bytes representin '0' in Uint32 format to indicate the length of input_type
+    //  output_type:undefined
+    // }  
+    // it has the same for 0x1000000010000000100000 (in ckb-sdk-js format)
+    // {
+    //  lock: undefined
+    //  input_type: undefined
+    //  output_type: undefined
+    // }  
+    expect(result).toEqual(HexUtils.byteLength('0x1400000010000000100000001400000000000000') + 4 + 4)
```

Comment:
It may cause here https://github.com/ckb-js/lumos/blob/develop/packages/codec/src/blockchain.ts#L64-L66 Lumos set all fields as `option`. But `ckb-sdk-js` does not set fields as `option`

Reply:
I feel confused with https://github.com/ckb-js/lumos/blob/develop/packages/codec/src/blockchain.ts#L64-L66.
When I call `bytes.hexify(base.blockchain.WitnessArgs.pack({ lock: '0x', input_type: '0x', output_type: '0x'}))`, it will output `0x1400000010000000140000001400000000000000`, why `input_type` and `output_type` set empty, but `lock` set '0x00000000'.

Reply:
here '0x00000000' is `args`, check the test case setup: https://github.com/nervosnetwork/neuron/pull/2734/files/dd6a4deaea055f8627e5e3362e7c87ef52ef9865#diff-67c688a92fedba703d332bc20e7149a30b2439001df088a9b6038b0b06d63470R19